torch/ao/quantization/fx/prepare.py

```diff
@@ -651,6 +687,8 @@ def maybe_insert_input_observers_for_node(
     qhandler: Optional[QuantizeHandler],
     prepare_custom_config: PrepareCustomConfig,
     backend_config: Optional[BackendConfig],
+    weight_index_dict: Dict[str, List[int]],
+    bias_index_dict: Dict[str, List[int]],
```

Comment:
By the way, I know you just followed the TODO comment but one optional feedback I have is maybe instead of passing these `weight_index_dict` and `bias_index_dict` everywhere, we should just pass `backend_config` around. This would simplify the function signatures a little and avoid introducing yet another data structure that the user has to know about. Then we can just check against `backend_config.configs[pattern]._input_type_to_index` in `node_arg_is_weight` and `node_arg_is_bias`.

We don't have to do this in this PR though. Feel free to merge as is.